webapp/store/app/hooks/index.js

```diff
@@ -1,48 +1,47 @@
-import { useState } from 'react'
-import { useSelector } from 'react-redux'
+import { useDispatch, useSelector } from 'react-redux'
 
 import * as AppState from '@webapp/app/appState'
 import { Area, Country } from '@common/country'
 import { useI18n, useOnUpdate } from '@webapp/components/hooks'
 
+import * as AppActions from '../actions'
+
 export const useAssessmentType = () => useSelector(AppState.getAssessmentType)
 
 const sortCountries = (countries, i18n) => {
   const compareListName = Area.getCompareListName(i18n)
   const compareCountries = (country1, country2) => compareListName(country1.countryIso, country2.countryIso)
-  return countries.sort(compareCountries)
+  return [...countries].sort(compareCountries)
 }
 
 export const useCountries = () => {
   const i18n = useI18n()
+  const dispatch = useDispatch()
   const countries = useSelector(AppState.getCountries)
 
-  const [countriesSorted, setCountriesSorted] = useState(() => sortCountries(countries, i18n))
-
   useOnUpdate(() => {
-    setCountriesSorted([...sortCountries(countries, i18n)])
-  }, [countries, i18n])
+    dispatch(AppActions.updateCountries(countries))
+  }, [i18n])
 
-  return countriesSorted
+  return sortCountries(countries, i18n)
```

Comment:
why do you sort all the time? the sorted countries should go in the store directly